### ISSUE TITLE
How to bind different input shape to executor in c++?

### ISSUE THREAD

**Issue by chenbiaolong** [apache/incubator-mxnet#8360]
I build a cnn network to extract images feature, however my input images do not have same shape,eg:
[batch, 1, 32, width], `batch` and `width` can change at each forward. so how can I bind different input data to an executor? One solution is rebind executor every time I run forward, but it cost too much time. Any example code solve this problem? my code now:
```
 int MXNetInference::GetOutputTensor(
                    const std::map<std::string, mxnet::cpp::NDArray>& inputs,
                    std::vector<mxnet::cpp::NDArray>& outputs) {
        for(auto& input: inputs) {
            args_map_[input.first] = input.second.Copy(*global_ctx_);
            NDArray::WaitAll();
        }
        /*bind the excutor*/
        executor_.reset(net_.SimpleBind(*global_ctx_, args_map_, map<string, NDArray>(),
                map<string, mxnet::cpp::OpReqType>(), aux_map_));
        executor_->Forward(false);
        /*print out the features*/
        std::vector<NDArray> results = executor_->outputs;
        for(auto& result: results) {
            NDArray array = result.Copy(Context(kCPU, 0));
            NDArray::WaitAll();
            outputs.push_back(array);
        }
        return 0;
    }
```

**Comment by reminisce** [comment]
Once binding finishes, shapes do not change at the inference stage. I am not aware of other approaches other than rebinding that are able to perform inference with different input shapes.

**Comment by chenbiaolong** [comment]
I use buckets to improve the inference performance.Building different executors binding with different input shape, and padding images to fill the buckets.Of course, buckets will do some useless compute and use more GPU memory,but I can't think out a better idea.

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by mike07026** [comment]
Yes, I meet the same problem when using mtcnn(face detector). I hope C/C++ interface support this case better. @szha
